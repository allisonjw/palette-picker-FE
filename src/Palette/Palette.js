import React from 'react';
import './Palette.scss';
import { useDispatch } from 'react-redux';
import trashBin from '../images/rubbish.png';
import { deletePalette } from '../util/apiCalls';
import { deleteColorPalette } from '../actions';

const Palette = ({palette_name, id, color_1, color_2, color_3, color_4, color_5}) => {
    const dispatch = useDispatch()
 
    const getColors = () => {
        const colorArray = [color_1, color_2, color_3, color_4, color_5]
        const makeColors = colorArray.map(color => {
            return(
                <div className="palette__colors" style={{backgroundColor: `${color}`}}>
                </div>
            )
        })
        return makeColors
    }

    const removePalette = (id) => {
        dispatch(deleteColorPalette(id));
        const body = {
          id: id
        }
        deletePalette('palettes', 'DELETE', body);
    };

    return (
        <section className="section_palettes">
            <h3 className="palette_name">{palette_name}</h3>
            {getColors()}
            <img
                className="palette_delete-btn"
                src={trashBin}
                alt="delete button"
                onClick={() => removePalette(id)}
            />
        </section>
    )
}

export default Palette;