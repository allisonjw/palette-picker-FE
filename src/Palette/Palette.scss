.section_palettes {
    background-color: white;
    display: flex;
    width: 285px;
    align-items: center;
    justify-content: space-evenly;
    border: solid 1px rgb(187, 187, 187);
    padding: 2px 10px 2px 5px;
    .section_palettes:last-child {
        border-radius: 0 0 15px 15px;
    }
}

.palette_name {
    width: 100px;
    text-align: left;
    line-height: 30px;
}

.palette_delete-btn {
    height: 25px;
    margin: 2px;
}

.palette__colors {
    height: 30px;
    width: 30px;
}