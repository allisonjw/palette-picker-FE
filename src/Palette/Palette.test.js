import React from 'react';
import { render } from '@testing-library/react';
import Palette from './Palette';
import { shallow } from 'enzyme';

describe('Palette', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Palette />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});