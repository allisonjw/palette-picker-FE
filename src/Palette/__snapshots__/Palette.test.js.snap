// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should match the snapshot 1`] = `
<section
  className="section_palettes"
>
  <h3
    className="palette_name"
  />
  <img
    alt="delete button"
    className="palette_delete-btn"
    src="rubbish.png"
  />
</section>
`;
